packages/axelar-soroban-std-derive/src/lib.rs

```diff
@@ -220,3 +222,119 @@ pub fn derive_upgradable(input: TokenStream) -> TokenStream {
         }
     }.into()
 }
+
+/// Implements the Event and EventTestUtils traits for a Soroban contract event.
+///
+/// Fields without a `#[data]` attribute are used as topics, while fields with `#[data]` are used as event data.
+/// The event name can be specified with `#[event_name(...)]` or will default to the struct name in snake_case (minus "Event" suffix).
+///
+/// # Example
+/// ```rust
+/// # use soroban_sdk::{String, Address};
+/// # use axelar_soroban_std_derive::IntoEvent;
+/// #[derive(Debug, PartialEq, IntoEvent)]
+/// pub struct TransferEvent { // event name is "transfer"
+///     pub from: Address,
+///     pub to: Address,
+///     #[data]
+///     pub amount: String,
+/// }
+/// ```
```

Comment:
show how to use this to emit an event in an example contract's entrypoint
https://github.com/axelarnetwork/axelar-cgp-soroban/blob/7da36774cfa4f44d7bd66de9ac04f8ed0d4c6160/packages/axelar-soroban-std-derive/src/lib.rs#L146